src/signalling.c

```diff
@@ -322,4 +338,28 @@ void handleUserInfo(PurpleConnection *gc, const char *who, PurpleNotifyUserInfo
     ev->freeMe = true;
     ev->data = msgData;
     signalling_push(ev);
+
+    return NULL;
 }
+
+void handleBuddyTyping(PurpleAccount *account, const char *name, void *data) {
+    handleTyping(account, name, true);
+}
+
+void handleBuddyTypingStopped(PurpleAccount *account, const char *name, void *data) {
+    handleTyping(account, name, false);
+}
+
+void handleTyping(PurpleAccount *account, const char *name, bool typing) {
+    s_signalEventData *ev = malloc(sizeof(s_signalEventData));
+    e_UserTyping *typingData = malloc(sizeof(e_UserTyping));
+    typingData->typing = typing;
+    typingData->account = account;
+    typingData->sender = malloc(strlen(name) + 1);
+    strcpy(typingData->sender, name);
```

Comment:
Maybe [`g_strdup`](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strdup) (or just its stdlib equivalent) so that you don't need to do the math? ;)